### PR TITLE
Migrate to new `BlackBoxBoundedContext` API

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR extends and modifies Kanban Board example. In particular:
  1. Adds `CardAlreadyPicked` rejection and its handling.
  2. Migrates to more compact event assertion API.
  3. Extracts abstract test suite bases.
  4. Encapsulates work with IDs in tests.
  5. Fixes placement of board-related proto events.
  6. Adds proto messages for `JoinCard` command, and `MemberJoinedCard` event. 
     Handling would be one in a following PR.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@mdrachuk, PTAL.

File: kanban-board/src/main/proto/spine_examples/kanban/card_rejections.proto [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// Putting rejection message classes under an outer class avoids name clash inside the package.
+//
+option java_multiple_files = false;
+
+option (type_url_prefix) = "type.spine.io";
+
+import "spine_examples/kanban/identifiers.proto";
+
+// There is already a member who picked the card before.
+message CardAlreadyPicked {
```
Comment (mdrachuk) [inline kanban-board/src/main/proto/spine_examples/kanban/card_rejections.proto]: Missing an empty doc line.

File: kanban-board/src/main/proto/spine_examples/kanban/card_rejections.proto [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    // The ID of the card which the actor wanted to pick.
```
Comment (mdrachuk) [inline kanban-board/src/main/proto/spine_examples/kanban/card_rejections.proto]: I don‘t see `actor` as a word from our domain. Can it be a user?
```suggestion
    // The ID of the card which the user wanted to pick.
```

File: kanban-board/src/main/proto/spine_examples/kanban/card_rejections.proto (hunk 1 above)
Comment (alexander-yevsyukov) [reply kanban-board/src/main/proto/spine_examples/kanban/card_rejections.proto]: I'm not sure we need to have them always. Let's continue experimenting with the layout, until we clearly see what works best.

**Pull request merged**